api/app/signals/apps/api/v1/views/pdf.py

```diff
@@ -74,6 +81,56 @@ def get_object(self):
         self.check_object_permissions(request=self.request, obj=obj)
         return obj
 
+    def _resize(self, image):
+        # Consider image orientation:
+        if image.width > image.height:
+            # landscape
+            width = self.max_size
+            height = int((self.max_size / image.width) * image.height)
+        else:
+            # portrait
+            width = int((self.max_size / image.height) * image.width)
+            height = self.max_size
+
+        return image.resize(size=(width, height), resample=Image.LANCZOS).convert('RGB')
+
+    def _get_context_data_images(self, signal):
+        jpg_data_urls = []
+        for att in signal.attachments.all():
+            # Attachment is_image property is currently not reliable
+            _, ext = os.path.splitext(att.file.name)
+            if ext not in ['.gif', '.jpg', '.jpeg', '.png']:
+                continue  # unsupported image format, or not image format
+
+            # Since we want a PDF to be output, we catch, log and ignore errors
+            # while opening attachments. A missing image is not as bad as a
+            # complete failure to render the requested PDF.
+            try:
+                with default_storage.open(att.file.name) as file:
+                    buffer = io.BytesIO(file.read())
+                image = Image.open(buffer)
+            except UnidentifiedImageError:
+                # PIL cannot open the attached file it is probably not an image.
+                msg = f'Cannot open image attachment pk={att.pk}'
```

Comment:
Perhaps make the message a bit more specific to the raised Exception. Not a blocker for now